src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/problem/ExceptionHandling.kt

```diff
@@ -21,6 +28,32 @@ class ExceptionHandling : ProblemHandling {
 
     return AdviceTraitDefault().toProblem(throwable, status)
   }
+
+  override fun handleMessageNotReadableException(
+    exception: HttpMessageNotReadableException,
+    request: NativeWebRequest
+  ): ResponseEntity<Problem> {
+    val responseBuilder = Problem.builder()
+      .withStatus(Status.BAD_REQUEST)
+      .withTitle("Bad Request")
+
+    if (exception.cause is InvalidFormatException) {
+      val invalidFormatException = exception.cause as InvalidFormatException
```

Comment:
I'm surprised you need this cast, I thought it would be smartcasted in the `if`.

Reply:
I think it's because smart casts only work on local variables or immutable members, otherwise you could get a race condition with another thread changing the value of cause